### ISSUE TITLE
Table not found error when checking the "API usage across geo location" in publisher analytics

### ISSUE THREAD

**Issue by nu1silva** [wso2/product-apim#1900]
**Description:**
Table not found error when checking the "API usage across geo location" in publisher analytics

**Suggested Labels:**
Bug

**Suggested Assignees:**
N/A

**Affected Product Version:**
APIM 2.1.0
IS as a KM 5.3.0
WUM updated

**OS, DB, other environment details and versions:**    
Ubuntu 14.04
MySQL 5.6

**Steps to reproduce:**
1. Setup APIM distributed setup with pattern-3 of [1] with IS as the KM
2. login to the publisher
3. Go to Analytics > API > API Usage Across Geo Locations
![image](https://user-images.githubusercontent.com/234736/33366631-d440b9d4-d511-11e7-8b41-63eb850cfbf7.png)

[1] https://github.com/wso2/puppet-apim

The following log was observed in the publisher logs.
Full log attatched.
```
TID: [-1234] [] [2017-11-29 08:55:38,573] ERROR {org.wso2.carbon.apimgt.usage.client.impl.APIUsageStatisticsRdbmsClientImpl} -  Error occurred while checking existence of the table:API_REQ_GEO_LOC_SUMMARY {org.wso2.carbon.apimgt.usage.client.impl.APIUsageStatisticsRdbmsClientImpl}
com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'CE_STAT_DB.API_REQ_GEO_LOC_SUMMARY' doesn't exist
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:404)
	at com.mysql.jdbc.Util.getInstance(Util.java:387)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:942)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3966)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3902)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2526)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2673)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2545)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2503)
	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1369)
	at sun.reflect.GeneratedMethodAccessor118.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tomcat.jdbc.pool.StatementFacade$StatementProxy.invoke(StatementFacade.java:114)
	at com.sun.proxy.$Proxy17.executeQuery(Unknown Source)
	at org.wso2.carbon.apimgt.usage.client.impl.APIUsageStatisticsRdbmsClientImpl.isTableExist(APIUsageStatisticsRdbmsClientImpl.java:2040)
	at org.wso2.carbon.apimgt.usage.client.impl.APIUsageStatisticsRdbmsClientImpl.getGeoLocationsByApi(APIUsageStatisticsRdbmsClientImpl.java:2772)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.mozilla.javascript.MemberBox.invoke(MemberBox.java:126)
	at org.mozilla.javascript.NativeJavaMethod.call(NativeJavaMethod.java:225)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.jaggeryjs.rhino.publisher.modules.statistics.c1._c_getGeolocationUsageByAPI_24(/publisher/modules/statistics/usage.jag:717)
	at org.jaggeryjs.rhino.publisher.modules.statistics.c1.call(/publisher/modules/statistics/usage.jag)
	at org.mozilla.javascript.ScriptRuntime.applyOrCall(ScriptRuntime.java:2430)
	at org.mozilla.javascript.BaseFunction.execIdCall(BaseFunction.java:269)
	at org.mozilla.javascript.IdFunctionObject.call(IdFunctionObject.java:97)
	at org.mozilla.javascript.optimizer.OptRuntime.call2(OptRuntime.java:42)
	at org.jaggeryjs.rhino.publisher.modules.statistics.c0._c_anonymous_27(/publisher/modules/statistics/module.jag:61)
	at org.jaggeryjs.rhino.publisher.modules.statistics.c0.call(/publisher/modules/statistics/module.jag)
	at org.mozilla.javascript.optimizer.OptRuntime.callN(OptRuntime.java:52)
	at org.jaggeryjs.rhino.publisher.site.blocks.stats.api_geolocation_usage.ajax.c0._c_anonymous_1(/publisher/site/blocks/stats/api-geolocation-usage/ajax/stats.jag:32)
	at org.jaggeryjs.rhino.publisher.site.blocks.stats.api_geolocation_usage.ajax.c0.call(/publisher/site/blocks/stats/api-geolocation-usage/ajax/stats.jag)
	at org.mozilla.javascript.optimizer.OptRuntime.call0(OptRuntime.java:23)
	at org.jaggeryjs.rhino.publisher.site.blocks.stats.api_geolocation_usage.ajax.c0._c_script_0(/publisher/site/blocks/stats/api-geolocation-usage/ajax/stats.jag:4)
	at org.jaggeryjs.rhino.publisher.site.blocks.stats.api_geolocation_usage.ajax.c0.call(/publisher/site/blocks/stats/api-geolocation-usage/ajax/stats.jag)
	at org.mozilla.javascript.ContextFactory.doTopCall(ContextFactory.java:394)
	at org.mozilla.javascript.ScriptRuntime.doTopCall(ScriptRuntime.java:3091)
	at org.jaggeryjs.rhino.publisher.site.blocks.stats.api_geolocation_usage.ajax.c0.call(/publisher/site/blocks/stats/api-geolocation-usage/ajax/stats.jag)
	at org.jaggeryjs.rhino.publisher.site.blocks.stats.api_geolocation_usage.ajax.c0.exec(/publisher/site/blocks/stats/api-geolocation-usage/ajax/stats.jag)
	at org.jaggeryjs.scriptengine.engine.RhinoEngine.execScript(RhinoEngine.java:567)
	at org.jaggeryjs.scriptengine.engine.RhinoEngine.exec(RhinoEngine.java:273)
	at org.jaggeryjs.jaggery.core.manager.WebAppManager.exec(WebAppManager.java:588)
	at org.jaggeryjs.jaggery.core.manager.WebAppManager.execute(WebAppManager.java:508)
	at org.jaggeryjs.jaggery.core.JaggeryServlet.doPost(JaggeryServlet.java:29)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:650)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:743)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:485)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:377)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:337)
	at org.jaggeryjs.jaggery.core.JaggeryFilter.doFilter(JaggeryFilter.java:21)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.wso2.carbon.ui.filters.cache.ContentTypeBasedCachePreventionFilter.doFilter(ContentTypeBasedCachePreventionFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:124)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:110)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:506)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:169)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.wso2.carbon.tomcat.ext.valves.CompositeValve.continueInvocation(CompositeValve.java:99)
	at org.wso2.carbon.tomcat.ext.valves.CarbonTomcatValve$1.invoke(CarbonTomcatValve.java:47)
	at org.wso2.carbon.webapp.mgt.TenantLazyLoaderValve.invoke(TenantLazyLoaderValve.java:57)
	at org.wso2.carbon.tomcat.ext.valves.TomcatValveContainer.invokeValves(TomcatValveContainer.java:47)
	at org.wso2.carbon.tomcat.ext.valves.CompositeValve.invoke(CompositeValve.java:62)
	at org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve.invoke(CarbonStuckThreadDetectionValve.java:159)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:962)
	at org.wso2.carbon.tomcat.ext.valves.CarbonContextCreatorValve.invoke(CarbonContextCreatorValve.java:57)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:445)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1115)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:637)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1775)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1734)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
TID: [-1234] [] [2017-11-29 08:55:38,584] ERROR {org.wso2.carbon.apimgt.usage.client.APIUsageStatisticsClient} -  Statistics Table:API_REQ_GEO_LOC_SUMMARY does not exist. {org.wso2.carbon.apimgt.usage.client.APIUsageStatisticsClient}
TID: [-1234] [] [2017-11-29 08:55:38,588] ERROR {JAGGERY.modules.statistics.usage:jag} -  org.wso2.carbon.apimgt.usage.client.exception.APIMgtUsageQueryServiceClientException: Statistics Table:API_REQ_GEO_LOC_SUMMARY does not exist. {JAGGERY.modules.statistics.usage:jag}
```

Full log:
[wso2carbon.log](https://github.com/wso2/product-apim/files/1513082/wso2carbon.log)

**Related Issues:**
N/A

**Comment by chamilaadhi** [comment]
Hi,
You need to specifically configure geo location based API stats [1]. If not, you will see this error.

[1]. https://docs.wso2.com/display/AM210/Configuring+Geolocation+Based+Statistics
